src/main/flight/pid.c

```diff
@@ -777,13 +778,20 @@ float pidGetAirmodeThrottleOffset(void)
 
 static FAST_CODE_NOINLINE void disarmOnImpact(void)
 {
-    // if all sticks are within 5% of center, and throttle low, check accDelta for impacts
-    // threshold should be high enough to avoid unwanted disarms in the air on throttle chops
-    if (isAirmodeActivated() && getMaxRcDeflectionAbs() < 0.05f && mixerGetRcThrottle() < 0.05f &&
-        fabsf(acc.accDelta) > pidRuntime.ezLandingDisarmThreshold) {
-        // disarm on accDelta transients
-        setArmingDisabled(ARMING_DISABLED_ARM_SWITCH);
+    // if, after takeoff...
+    if (isAirmodeActivated() && (
+#ifdef USE_ALTHOLD_MODE
+        // either altitude hold mode is active, including when activated by failsafe landing mode...
+        altHoldIsActive() ||
+#endif
+        // or throttle is near zero and sticks centered, and
+        (getMaxRcDeflectionAbs() < 0.05f && mixerGetRcThrottle() < 0.05f)) &&
```

Comment:
This I'm a bit uncertain about, generally.

It is easiest for LOS landings to put throttle where the descent is nice and slow, and leave it there.  Then, with the focus only on roll and pitch, some minor adjustments with the right hand stick can keep the quad directly over the landing point as it comes down slowly.  This is very helpful in a very tight landing spot.

In this case, the throttle position won't be zero and it won't disarm on touchdown.  If there is wind, the right stick may be offset to oppose it, and that can be enough to prevent the disarm.  And, to be honest, while testing it out, I would keep forgetting to put throttle to zero at least on some landings.

Hence the code above is really nice from the pilot's perspective, for LOS landings.

For a 'failsafe landing' mode, the stick positions set for landing mode will apply, and although they are centered for RPY, normally the default throttle value is non-zero. close to hover.
  
In both situations, ignoring the sticks makes the code a lot simpler.  

However, enabling the auto-disarm through the whole descent period does carry a risk that the quad could disarm from accelerometer noise or due to an impact during the descent.  A suitably high threshold (10, say) makes the former very unlikely, although with some extreme initiation scenarios the G forces could change quickly.  And an impact with a hard object at altitude is never pretty.

So one option to consider is to not ignore the sticks from the disarm perspective until, like GPS Rescue, we are below an altitude threshold (some height above the arming point).  That's how in GPS Rescue we can have quite a low threshold, and we bring it down the closer we get to ground.  

I don't have 100% clarity on the best approach, I guess.